app.ts

```diff
@@ -0,0 +1,175 @@
+
+async function placeRecordsFromCursor(cursor: number[]): Promise<number[]> {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return await placeRecords(cursor[0], cursor[1]);
+}
+//#endregion
+
+//#region Handlers
+async function getPageContent(fromID: number, toID: number): Promise<number[]> {
+    $("#WrapperTable-HeaderRow").empty();
+    getColumnNames()
+        .then((columns: string[]) => {
+            for (const column of columns) {
+                let writable: string = `<th align="center">${column}</th>`;
+                $("#WrapperTable-HeaderRow").append(writable);
+            }
+        });
+    return await placeRecords(fromID, toID);
+}
+
+function toNumber(input: string | number, output: any = 1) : number {
+    switch (typeof input) {
+        case ('string'):
+            if (output == 1) {
+                return parseInt(input as string);
+            } else {
+                return parseFloat(input as string);
+            }
+        case ("number"):
+            return input as number;
+        default:
+            return 0;
+    }
+}
+
+function calculateToId(fromId: number): number {
+    let recordDisplayOffset = 0;
+    if (window.innerHeight <= 646) {
+        recordDisplayOffset = 0
+    } else if (window.innerHeight <= 969) {
+        recordDisplayOffset = 1;
+    } else if (window.innerHeight <= 1938) {
+        recordDisplayOffset = 3
+    } else {
+        recordDisplayOffset = 15
+    }
+
+    return recordDisplayOffset + (fromId + Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37));
+}
+
+function nextPageResize(previousCursor: number[]): number {
+        const fromID = toNumber(previousCursor.sort((a, b) => {return a - b})[0]);
+        const toID = toNumber(previousCursor.sort((a, b) => {return a - b})[1]);
+        const documentHeight = ($(window).innerHeight() as number) - ($(`#table-row-${fromID.toString()}`).height() as number);
+
+        for (let i = fromID; i <= toID; i++) {
+            const elementHeightOffset = ($(`#table-row-${i.toString()}`).offset() as JQueryCoordinates).top as number;
+
+            if ( elementHeightOffset >= documentHeight ){
+                return i;
+            } else {
+                continue;
+            }
+        }
+        return toID;
+}
+
+function previousPageResize(previousCursor: number[]): number[] {
+    const toId = calculateToId(previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]));
+    return [previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]), toId];
+}
+//#endregion
+//#endregion
+
+let previousCursor: number[];
+let previousScale: number;
+
+window.onload = async () => {     
+    previousCursor = await getPageContent(0, calculateToId(0));
+    
+    $("#previous-page").click(async () => { 
+        previousCursor = previousPageResize(previousCursor);
+        let fromId = (previousCursor[0] >= 0 ? previousCursor[0] : 0);
```

Comment:
You don't need the brackets here and in some of your other ternaries

```suggestion
        let fromId = previousCursor[0] >= 0 ? previousCursor[0] : 0;
```